### PR TITLE
discord: Update activity message while playing PVP or DMM

### LINKED ISSUE #11430: Discord plugin leaks ingame activity wile playing Deadman Mode
Given the competitive PVP nature of DMM/SDMM/DMMT, we should not display the current activity of players on DMM. I think it would be safe to add a "Playing Deaman Mode" activity line rather than omitting it like we do for wilderness activities.

### PR DESCRIPTION
**Pull request by rangvaldr**
Closes #11430.

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/WorldType.java [hunk 1]
```diff
@@ -77,10 +78,13 @@
 	}
 
 	private static final EnumSet<WorldType> PVP_WORLD_TYPES = EnumSet.of(
-		DEADMAN,
```
Comment (Nightfirecat) [inline runelite-api/src/main/java/net/runelite/api/WorldType.java:80]: Deadman is still a PVP world type. Why is this removed?

File: runelite-api/src/main/java/net/runelite/api/WorldType.java [hunk 2]
```diff
@@ -131,4 +135,16 @@ public static boolean isPvpWorld(final Collection<WorldType> worldTypes)
...
 	}
+
+	/**
+	 * Checks whether a world having a {@link Collection} of {@link WorldType}s is a Deadman world.
+	 *
+	 * @param worldTypes A {@link Collection} of {@link WorldType}s describing the given world.
+	 * @return           True if the given worldtypes of the world are a Deadman world, false otherwise.
+	 * @see Client#getWorldType()
+	 */
+	public static boolean isDeadmanWorld(final Collection<WorldType> worldTypes)
```
Comment (Nightfirecat) [inline runelite-api/src/main/java/net/runelite/api/WorldType.java:146]: This can be replaced by `Client.getWorldType().contains(WorldType.DEADMAN)` since there is only one deadman world type.

File: runelite-api/src/main/java/net/runelite/api/WorldType.java (hunk 1 above)
Comment (rangvaldr) [reply runelite-api/src/main/java/net/runelite/api/WorldType.java]: Well it can be left, but the deadman check comes before the PVP world check. So if they're in a deadman world, `PVP_WORLD_TYPES` will never be checked.

Comment (Nightfirecat) [reply runelite-api/src/main/java/net/runelite/api/WorldType.java]: `isPvpWorld` is used outside of the discord plugin, so this would cause adverse effects for other plugins.

Comment (rangvaldr) [reply runelite-api/src/main/java/net/runelite/api/WorldType.java]: Fixed.

File: runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java [hunk 3]
```diff
@@ -381,6 +381,12 @@ private void checkForAreaUpdate()
 			return;
 		}
 
+		if (client.getWorldType().contains(WorldType.DEADMAN))
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java:384]: Why are you implementing this differently than current PVP world check? Just use the same logic and in same method

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java:384]: Actually nvm, that is already contained in PVP check. I don't understand why anything extra needs to be implemented

Comment (rangvaldr) [reply runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java:384]: Well if you were okay with just hiding the activity completely, similar to PVP, no changes would be necessary. But in #11430 you had mentioned adding the activity line "Playing Deadman Mode", which is what's implemented here and does require the additional changes. If you just want to hide the activity, feel free to close the PR.

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java:384]: Imo it is fine to display playing DMM and I would be happy merging that, my only requirement is for it to be implemented for all cases we are skipping atm (e.g all types of PVP worlds)

Comment (rangvaldr) [reply runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java:384]: Gotcha.

Comment (rangvaldr) [reply runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java]: Resolved with 30eab1a.

File: runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java [hunk 4]
```diff
@@ -381,6 +381,19 @@ private void checkForAreaUpdate()
 			return;
 		}
 
+		final EnumSet<WorldType> worldType = client.getWorldType();
+
+		if (worldType.contains(WorldType.DEADMAN))
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java:386]: this should return playing DMM even for tournament DMM worlds imo

Comment (rangvaldr) [reply runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java:386]: There was a comment in the code that tournament worlds were also flagged deadman.

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordPlugin.java:386]: Nvm, I just read the comment in PVP_WORLD_TYPES, this is fine
